env0/resource_environment.go

```diff
@@ -156,6 +272,84 @@ func getUpdatePayload(d *schema.ResourceData) client.EnvironmentUpdate {
 	return payload
 }
 
+func getDeployPayload(d *schema.ResourceData) client.DeployRequest {
+	payload := client.DeployRequest{}
+
+	if templateId, ok := d.GetOk("template_id"); ok {
+		payload.BlueprintId = templateId.(string)
+	}
+
+	if revision, ok := d.GetOk("revision"); ok {
+		payload.BlueprintRevision = revision.(string)
+	}
+
+	if configuration, ok := d.GetOk("configuration"); ok {
+		configurationChanges := getConfigurationVariables(configuration.([]interface{}))
+		payload.ConfigurationChanges = &configurationChanges
+	}
+
+	if ttl, ok := d.GetOk("ttl"); ok {
+		payload.TTL = &client.TTL{
+			Type:  ttl.(map[string]interface{})["type"].(string),
+			Value: ttl.(map[string]interface{})["value"].(string),
+		}
+	}
+
+	if userRequiresApproval, ok := d.GetOk("requires_approval"); ok {
+		payload.UserRequiresApproval = userRequiresApproval.(bool)
+	}
+
+	return payload
+}
+
+func getConfigurationVariables(configuration []interface{}) client.ConfigurationChanges {
+	configurationChanges := client.ConfigurationChanges{}
+	for _, variable := range configuration {
+		configurationVariable := getConfigurationVariableForEnvironment(variable.(map[string]interface{}))
+		configurationChanges = append(configurationChanges, configurationVariable)
+	}
+	return configurationChanges
+}
+
+func getConfigurationVariableForEnvironment(variable map[string]interface{}) client.ConfigurationVariable {
+	varType := VariableTypes[variable["type"].(string)]
+
+	configurationVariable := client.ConfigurationVariable{
+		Name:  variable["name"].(string),
+		Value: variable["value"].(string),
+		Scope: client.Scope(variable["scope"].(string)),
```

Comment:
i'd rather us to use the same convention from the resource
where we infer the scope and the scope id from what the user has given us

Reply:
I don't understand
thats how the api expects it wdym infer it like the resource the variable resource?